unittests/runtime/Concurrent.cpp

```diff
@@ -42,3 +44,48 @@ TEST(ConcurrentReadableArrayTest, SingleThreaded) {
   add(1000000);
   check();
 }
+
+TEST(ConcurrentReadableArrayTest, MultiThreaded) {
+  const int writerCount = 16;
+  const int readerCount = 8;
+  const int insertCount = 100000;
+
+  struct Value {
+    int threadNumber;
+    int x;
+  };
+  ConcurrentReadableArray<Value> array;
```

Comment:
Looks like the only write operation this exercises is `push_back`.  Do array elements get mutated?  Is that something that we should test?

Reply:
`push_back` is the only mutation this type supports currently. It's very niche.